scipy/spatial/tests/test_qhull.py

```diff
@@ -701,6 +701,120 @@ def test_volume_area(self):
         assert_allclose(tri.volume, 1., rtol=1e-14)
         assert_allclose(tri.area, 6., rtol=1e-14)
 
+    @pytest.mark.parametrize("incremental", [False, True])
+    def test_good2d(self, incremental):
+        # Make sure the QGn option gives the correct value of "good".
+        points = np.array([[0.2, 0.2],
+                           [0.2, 0.4],
+                           [0.4, 0.4],
+                           [0.4, 0.2],
+                           [0.3, 0.6]])
+        hull = qhull.ConvexHull(points=points,
+                                incremental=incremental,
+                                qhull_options='QG4')
+        expected = np.array([False, True, False, False], dtype=bool)
+        actual = hull.good
+        assert_equal(actual, expected)
+
+    @pytest.mark.parametrize("visibility", [
+                              "QG4",  # visible=True
+                              "QG-4",  # visible=False
+                              ])
+    @pytest.mark.parametrize("new_gen, expected", [
+        # add generator that places QG4 inside hull
+        # so all facets are invisible
+        (np.array([[0.3, 0.7]]),
+         np.array([False, False, False, False, False], dtype=bool)),
+        # adding a generator on the opposite side of the square
+        # should preserve the single visible facet & add one invisible
+        # facet
+        (np.array([[0.3, -0.7]]),
+         np.array([False, True, False, False, False], dtype=bool)),
+        # split the visible facet on top of the square into two
+        # visible facets, with visibility at the end of the array
+        # because add_points concatenates
+        (np.array([[0.3, 0.41]]),
+         np.array([False, False, False, True, True], dtype=bool)),
+        # with our current Qhull options, coplanarity will not count
+        # for visibility; this case shifts one visible & one invisible
+        # facet & adds a coplanar facet
+        # simplex at index position 2 is the shifted visible facet
+        # the final simplex is the coplanar facet
+        (np.array([[0.5, 0.6], [0.6, 0.6]]),
+         np.array([False, False, True, False, False], dtype=bool)),
+        # degeneracy will also count as invisibility with current
+        # Qhull config
+        (np.array([[0.3, 0.6]]),
```

Comment:
This is the problem case -- adding the degenerate generator incrementally fails only on 32-bit Linux.

Reply:
My current attempted fix here (which seems to work in a 32-bit Docker container) is to add `1e-16` to the y-value such that the query point is just barely inside the Convex Hull formed by adding this new point, near the limit of double precision.

It seems, albeit speculatively, that *exact* degeneracy may be handled slightly differently on 32-bit vs. 64-bit POSIX platforms, even with `m32` compile flag addition. It could be that our default usage of `Qt` Qhull option for all scenarios is the culprit on that, but that's generally a desirable option & I don't think we should add a special hook to disable graceful degeneracy handling for this obscure scenario ("is something visible if you're exactly on top of it?").

So my suggestion is that as far as visibility is concerned, if you add a generator that is *exactly* degenerate with your query point, it is fair game for Qhull to effectively move that point a bit and return **visibility** results that may vary on different architectures.

Maybe a concise doc addition is warranted, but on the other hand we already document that `Qt` is **always** enabled.